src/process/ProcessManagerImpl.cpp

```diff
@@ -486,6 +486,7 @@ ProcessManagerImpl::maybeRunPendingProcesses()
         }
         catch (std::runtime_error& e)
         {
+            i->cancel(std::make_error_code(std::errc::io_error));
```

Comment:
it looks like both versions (posix and windows) of `ProcessExitEvent::Impl::run()` are buggy and don't clean up all their resources when throwing (posix leaks `fileActions` and windows leaks `hStdOutput` on the second throw), we should fix this as part of this PR